### PR TITLE
Treat all warnings as errors by default

### PR DESCRIPTION
**Pull request by kubkon**
cc @peterhuene

### CODE REVIEW

**Comment by marmistrz** [comment]
I think we should rather only treat warnings as errors during CI and allow them for usual builds (e.g. during prototyping)

**Comment by kubkon** [comment]
> I think we should rather only treat warnings as errors during CI and allow them for usual builds (e.g. during prototyping)

That's a fair point. Let me fix it then.

**Comment by peterhuene** [comment]
I also agree that we should probably limit this to CI as warnings are pretty frequent during the development process.

File: src/fdentry.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 use crate::sys::fdentry_impl::{determine_type_and_access_rights, OsFile};
-use crate::{host, Error, Result};
+use crate::{host, wasm32, Error, Result};
```
Comment (kubkon) [inline src/fdentry.rs]: This one is just a temp change to see if the CI detects this correctly and fails. It will be removed before merging :-)

**Comment by sunfishcode** [comment]
Is this enabling clippy, and requiring that all PRs be clippy-clean?

I'm somewhat skeptical of that approach. While the warnings builtin into rustc are carefully crafted to be unambiguously worth fixing, a lot of clippy's warnings are quite subjective, and in other projects, I've seen it become a nuisance, especially when its warnings change as clippy evolves.

**Comment by peterhuene** [comment]
It would, yes.  If we don't want that, perhaps (assuming I have the GitHub actions correct):

```yaml
...
    - name: Build debug
      env:
        RUSTFLAGS: "-D warnings"
      uses: actions-rs/cargo@v1
      with:
        command: build
    - name: Build release
      uses: actions-rs/cargo@v1
      env:
        RUSTFLAGS: "-D warnings"
      with:
        command: build
        args: --release
...
```

**Comment by kubkon** [comment]
I guess it’s worth a try, thanks @peterhuene. I reckon that passing the flags to the compiler only in debug build should be more than enough though.

**Review by peterhuene — APPROVED** [review]

**Comment by marmistrz** [comment]
Is there any rationale for doing both debug and release build in the CI?

**Comment by kubkon** [comment]
> Is there any rationale for doing both debug and release build in the CI?

Good question. I guess we don't really need the debug build especially since we're running tests later on anyhow (albeit with a couple more dependencies than if invoking `cargo build`). Anyhow, I'm happy to strip away the debug build :-)

**Pull request merged**